WORKSPACE

```diff
@@ -220,20 +229,69 @@ node_repositories(
     yarn_version = "1.22.11",
 )
 
-# install external dependencies for pkg/ui package
+load("//build/bazelutil:seed_yarn_cache.bzl", "seed_yarn_cache")
+seed_yarn_cache(name = "yarn_cache")
+
+# Install external dependencies for NPM packages in pkg/ui/ as separate bazel
+# repositories, to avoid version conflicts between those packages.
+# Unfortunately Bazel's rules_nodejs does not support yarn workspaces, so 
+# packages have isolated dependencies and must be installed as isolated
+# Bazel repositories.
+yarn_install(
+    name = "npm_protos",
+    args = [
+        "--offline",
+        "--ignore-optional",
+    ],
+    data = [
+      "//pkg/ui:.yarnrc",
+      "@yarn_cache//:protos.seed",
+    ],
+    package_path = "/",
+    package_json = "//pkg/ui/workspaces/db-console/src/js:package.json",
+    strict_visibility = False,
+    yarn_lock = "//pkg/ui/workspaces/db-console/src/js:yarn.lock",
+)
+
 yarn_install(
-    name = "npm",
+    name = "npm_db_console",
     args = [
+        "--offline",
         "--ignore-optional",
+    ],
+    data = [
+      "//pkg/ui:.yarnrc",
+      "@yarn_cache//:db_console.seed",
```

Comment:
Nit: It seems like the 3 files you're exposing from this `yarn_cache` repo don't actually have separate meanings -- whether you depend on `db_console.seed` or `cluster_ui.seed` or whatever doesn't make a difference. You might as well have one file called `.seed`. OR you can have 3 different repositories, each of which seeds one directory (`@yarn_seed_db_console//:.seed`, `@yarn_seed_cluster_ui//:.seed`), and then which you choose for the data dependency would be meaningful. The current in-between state is kind of confusing though.

Reply:
Good catch!  I was hoping to keep the various seeds isolated, so that building `cluster-ui` doesn't require `db-console` dependencies to be seeded.  Should be easy to split this `repository_rule` into a "one per package" kind of thing :)